### PR TITLE
Crud user

### PR DESCRIPTION
**Pull request by alexamsv**

### CODE REVIEW

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java [hunk 1]
```diff
@@ -94,6 +96,10 @@ public boolean isCredentialsNonExpired() {
     public boolean isEnabled() {
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:96]: this is the same as IsActive.. you can remove this property as we don't need it anymore

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java [hunk 2]
```diff
@@ -47,17 +48,18 @@ public class User extends BaseEntity implements UserDetails {
     @Column(name = "description")
     private String description;
 
-    @Lob
-    @Column(name = "picture")
-    private byte[] picture;
+    @Column(name = "is_active")
+    private boolean isActive;
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:52]: Set default value isActive=true

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java [hunk 3]
```diff
@@ -94,6 +96,10 @@ public boolean isCredentialsNonExpired() {
     public boolean isEnabled() {
         return true;
     }
+
+    public boolean setActive(boolean b) {
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:100]: why do we need this setActive?

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+
+    @Override
+    public List<UserResponse> getAllUsers() {
+        List<User> users = userRepository.findAllByIsActive(true);
+        return users.stream()
+                .map(UserMapper.INSTANCE::userToUserResponse)
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:29]: use the injected mapper

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                .collect(Collectors.toList());
+    }
+
+
+    @Override
+    public UserResponse getUserById(Integer userId) {
+        User user = userRepository.findById(userId).orElse(null);
+        return user != null ? UserMapper.INSTANCE.userToUserResponse(user) : null;
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:37]: use the injected mapper

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+
+    @Override
+    public UserResponse updateUser(Integer userId, UserUpdateRequest userToUpdate) {
+        User existingUser = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+        UserMapper.INSTANCE.updateUserFromRequest(userToUpdate, existingUser);
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:45]: use the injected mapper everywhere, we dont need this INSTANCE

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        User updatedUser = userRepository.save(existingUser);
```
Comment (DameStoilovski99) [inline Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:46]: fix this logic, you dont save the updatedUser, test it!

File: Backend/src/main/java/com/service/marketplace/controller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            } else {
+                return ResponseEntity.ok(updatedUser);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            return ResponseEntity.internalServerError().build();
+        }
+    }
+
+    @DeleteMapping("/delete/{userId}")
```
[the remark this answers is not in the document]
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/controller/UserController.java:62]: > here the same, leave only the /{userId}

okay

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java (hunk 1 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:96]: > this is the same as IsActive.. you can remove this property as we don't need it anymore

okay

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java (hunk 2 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:52]: > Set default value isActive=true

okay

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java (hunk 5 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:37]: > use the injected mapper

okay

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:45]: > use the injected mapper everywhere, we dont need this INSTANCE

okay

File: Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java (hunk 7 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/service/impl/UserServiceImpl.java:46]: > fix this logic, you dont save the updatedUser, test it!

okay

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java (hunk 3 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:100]: > why do we need this setActive?

I will remove it.

File: Backend/src/main/java/com/service/marketplace/persistence/entity/User.java (hunk 1 above)
Comment (alexamsv) [reply Backend/src/main/java/com/service/marketplace/persistence/entity/User.java:96]: > this is the same as IsActive.. you can remove this property as we don't need it anymore

The isEnabled() method is part of the UserDetails interface, and attempting to remove it from our User class would result in an error due to the read-only nature of the file.

**Review by DameStoilovski99 — APPROVED** [review]

**Pull request merged**